sklearn/feature_selection/from_model.py

```diff
@@ -199,9 +215,33 @@ def _get_support_mask(self):
                 'Either fit the model before transform or set "prefit=True"'
                 ' while passing the fitted estimator to the constructor.')
         scores = _get_feature_importances(estimator)
-        self.threshold_ = _calculate_threshold(estimator, scores,
-                                               self.threshold)
-        return scores >= self.threshold_
+        mask = np.zeros(scores.shape, dtype=bool)
+        if self.threshold is not None and self.max_n_features is not None:
+            self.threshold_ = _calculate_threshold(estimator, scores,
+                                                   self.threshold)
+            mask = scores >= self.threshold_
+            if self.max_n_features == 'all':
+                pass
+            elif self.max_n_features == 0:
+                mask[:] = False
+            else:
+                scores = np.ma.masked_array(scores, mask=1-mask)
```

Comment:
What's wrong with something like

```python
order = np.argsort(scores, kind='mergesort')
mask[order[:-self.max_n_features]] = False
```
?

The point is that from the argsort, the threshold mask will cover a suffix, as will the `max_n_features` mask.